addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/CorsScanRule.java

```diff
@@ -112,8 +112,16 @@ public void scan() {
                         respHead.getHeader(HttpFieldsNames.ACCESS_CONTROL_ALLOW_CREDENTIALS);
                 acacVal = acacVal == null ? "" : acacVal;
 
+                // Check if the response indicates an error (4xx or 5xx status codes)
+                // In such cases, sensitive content is not actually being exposed
+                int statusCode = respHead.getStatusCode();
+                boolean isErrorResponse = statusCode >= 400;
+
                 // Evaluates the risk for this alert
-                if (acaoVal.contains("*")) {
+                if (isErrorResponse) {
+                    // Error responses don't expose sensitive content, keep risk as INFO
```

Comment:
Are you sure?
How can you be sure?
If there is sensitive content is there less risk/sensitivity to the finding?

Shouldn't this be modifying confidence not risk?